### ISSUE TITLE
Closing electron app is not clearing password auth cache

### ISSUE THREAD

**Issue by manikantag** [electron/electron#8233]
* Electron version: 1.4.12
* Operating system: Windows 8.1 x64
* AngularJS 1.5.8

### Expected behavior

Closing the app should clear the basic auth cahce. At least that's what the Chrome & other browsers are doing.

### Actual behavior

Closing the app and relaunching it still has the basic auth cache intact. In fact, the `login` event listener is not being called.

### How to reproduce

Code is very trivial and grabbed directly from the docs.

    app.on('login', (event, webContents, request, authInfo, callback) => {
        event.preventDefault();
        callback('someusername', 'somepassword');
    })

And when I m requesting a http endpoint which is protected by `basic` auth, it is still using old username & password from previous runs and the above `login` event listener is not called at all once it is called. In other words, the event listener is called once and once the auth is success, it is not getting called even after the app is closed and re-launched.

I had to explicitly clear the auth cache on app launch. Code I've to clear auth cache., which again took so much time to figure out the correct way of doing it (clue: it should be done from Renderer process. At least for me, the same below code didn't work from Main process).

    /**
     * Clears Electron's password auth cache.
     * http://electron.atom.io/docs/api/session/#sesclearauthcacheoptions-callback
     */
    function clearElectronPasswordAuthCache() {
        const remote = require('electron').remote;
        // Do this at the end of the current call stack as
        // somehow auth cache is not being cleared if not.
        setTimeout(function () {
            remote.session.defaultSession.clearAuthCache({ type: 'password' }, function () {
                console.log('Electron password auth cache cleared');
            })

            // Other way of clearing auth cache.
            // remote.getCurrentWindow().webContents.session.clearAuthCache({ type: 'password' }, function () {
            //     console.log('Electron password auth cache cleared');
            // })
        });
    }

I m calling this method in JS file which is included in main html. And the interesting thing is, I had to do this in `setTimeout`, which otherwise has no effect. So, if I didn't call the above method, still the old username & password from previous successful auth are being used.

I should also mention, I m running with `electron .`, and after installing the app. Does that make any difference? *I've not tested by installing, though.*

Is this the behavior? Or am I missing anything? Thanks.

**Comment by deepak1556** [comment]
Can you provide a minimal example to reproduce the issue ? I am unable to reproduce with https://httpbin.org/basic-auth/user/passwd . It would be great if you can include the behavior of `clearAuthCache` with the example.

**Comment by manikantag** [comment]
My bad :( 

It is working as expected: closing the app **is clearing** the password auth cache. I figured it out after trying to reproduce it with electron quick start app, and I couldn't reproduce it.

Problem with my app is, I m giving correct credentials for websocket connection auth, i.e., I've websocket URL as `wss://someusername:somepassword@server.com` (I m using URL like this as Chrome doesn't support auth for WebSockets *- correct me if I m wrong!*). So, when Electron/Chrome established a successful WS connection, it is remembering the creds, which is expected. And same creds are being used for all other Ajax calls too, which is obvious again. Thanks @deepak1556 for asking me to reproduce it 👍 

> It would be great if you can include the behavior of clearAuthCache with the example.

As I mentioned above, the WS connection is happening after the cache clear and thus using `setTimeout` is postponing the cache clear after the WS connection. But it is not required, and working as expected.

I've other doubt: how to call the **login** `callback` from Renderer process? Because in my application, end user login credentials will be validated by server request by sending the user given credentials as basic auth, which will happen in AngularJS code (i.e., Renderer process). I tried with `ipcMain` and `ipcRenderer`, but is not working as I can't pass functions when sending messages from main to renderer. And also I m not able to send username & password from renderer to main synchronously. Is there any way of doing this?

I m closing this issue anyway, Thanks.
